sound/soc/bcm/hifiberry_dacplus.c

```diff
@@ -301,9 +311,69 @@ static struct snd_soc_card snd_rpi_hifiberry_dacplus = {
 	.num_links    = ARRAY_SIZE(snd_rpi_hifiberry_dacplus_dai),
 };
 
+static int hb_hp_detect(void)
+{
+	struct i2c_adapter *adap = i2c_get_adapter(1);
+	int ret;
+
+	struct i2c_client tpa_i2c_client = {
+		.addr = 0x60,
+		.adapter = adap,
+	};
+
+	ret = i2c_smbus_read_byte(&tpa_i2c_client) >= 0;
+	i2c_put_adapter(adap);
+	return ret;
+};
+
+static struct property tpa_enable_prop = {
+	       .name = "status",
+	       .length = sizeof("okay") + 1,
```

Comment:
This line only works by accident, and only in 32-bit builds. `"okay"` is a `const char *`, which occupies 4 bytes - the same as the string length. In a 64-bit build the sizeof would have the value 8. Just write `4 + 1`.

Reply:
This is a good one - agreed! Thanks! Will fix...

Reply:
Actually, a string literal is an array of size N const chars, that decays to const char pointer in most uses.
But that doesn't include `sizeof`. `sizeof("okay")` is 5 (on 32 or 64-bit) - it will include the terminating null.
So correct fix is to remove the `+1`. (although '4+1' also works).